### ISSUE TITLE
RightShift lexer rule eclipsing double template end character '>'

### ISSUE THREAD

**Issue by romeo-cmd** [antlr/grammars-v4#1753]
I think lexer rule `RightShift : '>>'` unables matching rule `simpletemplateid` in the case where there are more than two levels of imbrication of `simpletemplateid` rule.

For instance, the explicit template instanciation `template class MyTemplate1<MyTemplate2<MyClass>>;` will not be recognized because symbol `>>` is matched with lexeme `RightShift`.

To correct it, I propose to change lexer rule `RightShift : >>` into parser rule `rightshift : '>' '>'` and change every reference to rule `RightShift` into a reference to rule `rightshift`.

Note: This problem might also happen with lexemes `RightShiftAssign`, `LeftShift`, and `LeftShiftAssign` but I think in practice it is impossible.

**Comment by KvanTTT** [comment]
Do you mean C++ grammar?

**Comment by romeo-cmd** [comment]
Yes, sorry.

**Comment by KvanTTT** [comment]
Yes, it makes sense. We have the same tokens for [java](https://github.com/antlr/grammars-v4/blob/master/java/java/JavaParser.g4#L487), [Objective-C](https://github.com/antlr/grammars-v4/blob/master/java/java/JavaParser.g4#L487) grammars.

**Comment by KvanTTT** [comment]
Would you like to suggest a pull request with the fix?

**Comment by romeo-cmd** [comment]
Yes why not! How can I do it?

**Comment by KvanTTT** [comment]
See instructions here: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests
